### PR TITLE
TF-1600 No notification for content of `Draft, Sent, Outbox, Spam, Trash`

### PR DESCRIPTION
**Pull request by dab246**
### Issue

#1600 

### Resolved

https://user-images.githubusercontent.com/80730648/228835213-e68c0724-6795-4304-8f5d-bbfacb356871.mp4

### CODE REVIEW

File: lib/features/push_notification/data/repository/fcm_repository_impl.dart [hunk 1]
```diff
@@ -104,4 +110,26 @@ class FCMRepositoryImpl extends FCMRepository {
...
+  @override
+  Future<List<PresentationMailbox>> getMailboxesNotPutNotifications(Session session, AccountId accountId) async {
+    final mailboxesCache = await _mapMailboxDataSource[DataSourceType.local]!.getAllMailboxCache();
+    final mailboxesCacheNotPutNotifications = mailboxesCache
+      .map((mailbox) => mailbox.toPresentationMailbox())
+      .where((presentationMailbox) => presentationMailbox.pushNotificationDeactivated)
+      .toList();
+    log('FCMRepositoryImpl::getMailboxesNotPutNotifications():mailboxesCacheNotPutNotifications: $mailboxesCacheNotPutNotifications');
+    if (mailboxesCacheNotPutNotifications.isNotEmpty && mailboxesCacheNotPutNotifications.length == 5) {
+      return mailboxesCacheNotPutNotifications;
```
Comment (nqhhdev) [inline lib/features/push_notification/data/repository/fcm_repository_impl.dart:123]: Why did you use the condition `mailboxesCacheNotPutNotifications.length == 5`? Let me knowwww ạaa

Comment (hoangdat) [reply lib/features/push_notification/data/repository/fcm_repository_impl.dart:123]: can you write the ADR?

Comment (dab246) [reply lib/features/push_notification/data/repository/fcm_repository_impl.dart]: > can you write the ADR?

Updated!. Please see below.

**Review by nqhhdev — APPROVED** [review]
lgtm

File: model/lib/extensions/presentation_email_extension.dart [hunk 2]
```diff
@@ -243,4 +244,19 @@ extension PresentationEmailExtension on PresentationEmail {
...
+  bool isBelongToOneOfTheMailboxes(List<MailboxId> mailboxIdsSource) {
+    final mapMailboxIds = mailboxIds;
+    mapMailboxIds?.removeWhere((key, value) => !value);
+
+    if (mapMailboxIds?.isNotEmpty == true) {
+      final listMailboxId = mapMailboxIds!.keys.toList();
+      log('PresentationEmailExtension::isBelongToOneOfTheMailboxes():listMailboxId: $listMailboxId');
+      final listMailboxIdValid = listMailboxId.where((mailboxId) => mailboxIdsSource.contains(mailboxId));
+      log('PresentationEmailExtension::isBelongToOneOfTheMailboxes():listMailboxIdValid: $listMailboxIdValid');
+      return listMailboxIdValid.isNotEmpty;
```
Comment (hoangdat) [inline model/lib/extensions/presentation_email_extension.dart:257]: How about?
```suggestion
      return mapMailboxIds!.keys.where((mailboxId) => mailboxIdsSource.contains(mailboxId)).isNotEmpty
```

Comment (dab246) [reply model/lib/extensions/presentation_email_extension.dart:257]: > How about?

I wanted it to make more sense to the reader, so I wrote it like that. Do you find it confusing? @hoangdat

**Review by hoangdat — APPROVED** [review]

**Pull request merged**